cardinal/util.go

```diff
@@ -48,3 +52,34 @@ func isFatalError(err error) bool {
 	}
 	return true
 }
+
+func GetMessage[In any, Out any](wCtx Context) (*MessageType[In, Out], error) {
+	var msg MessageType[In, Out]
+	msgType := reflect.TypeOf(msg)
+	tempRes, ok := wCtx.getMessageByType(msgType)
+	if !ok {
+		return nil, eris.Errorf("Could not find %q, Message may not be registered.", msg.Name())
+	}
+	var _ types.Message = &msg
+	res, ok := tempRes.(*MessageType[In, Out])
+	if !ok {
+		return &msg, eris.New("wrong type")
+	}
+	return res, nil
+}
+
+func GetTransactionReceiptsForTick(wCtx Context, tick uint64) ([]receipt.Receipt, error) {
```

Comment:
Unclear why you can't use `cardinal.EachMessage` instead

Reply:
As stated above. Tests that already used internal methods, I just wrapped those methods in public functions as they got privatized. I'll change these to use the alternative patterns you mentioned tmrw.